buttercup/cogs/stats.py

```diff
@@ -42,6 +48,121 @@ async def _stats(self, ctx: SlashContext) -> None:
             embed=Embed(title=i18n["stats"]["embed_title"], description=description),
         )
 
+    @cog_ext.cog_slash(
+        name="progress",
+        description="Get the transcribing progress of a user.",
+        options=[
+            create_option(
+                name="username",
+                description="The user to get the progress of. "
+                "Defaults to the user executing the command.",
+                option_type=3,
+                required=False,
+            ),
+            create_option(
+                name="hours",
+                description="The time frame of the progress in hours. "
+                "Defaults to 24 hours.",
+                option_type=4,
+                required=False,
+            ),
+        ],
+    )
+    async def _progress(
+        self, ctx: SlashContext, username: Optional[str] = None, hours: int = 24,
+    ) -> None:
+        """Get the transcribing progress of a user in the given time frame."""
+        start = datetime.now()
+        user = username or extract_username(ctx.author.display_name)
+        # Send a first message to show that the bot is responsive.
+        # We will edit this message later with the actual content.
+        msg = await ctx.send(
+            i18n["progress"]["getting_progress"].format(user=user, hours=hours)
+        )
+
+        volunteer_response = self.blossom_api.get_user(user)
+        if volunteer_response.status != BlossomStatus.ok:
+            await msg.edit(content=i18n["progress"]["user_not_found"].format(user))
+            return
+        volunteer_id = volunteer_response.data["id"]
+
+        if volunteer_response.data["gamma"] == 0:
+            # The user has not started transcribing yet
+            await msg.edit(
+                content=i18n["progress"]["embed_message"].format(
+                    get_duration_str(start)
+                ),
+                embed=Embed(
+                    title=i18n["progress"]["embed_title"].format(user),
+                    description=i18n["progress"]["embed_description_new"].format(
+                        user=user
+                    ),
+                ),
+            )
+            return
+
+        one_day_ago = start - timedelta(hours=hours)

```

Comment:
This is not necessarily a day ago, as it is an argument

Reply:
Ah right, I forgot about this when I added this parameter